[package] arrow/10.0.0: undefined references to re2

### Description

trying to use arrow with some parquet example code results in undefined references to re2 symbols

### Package and Environment Details

* Package Name/Version: **arrow/10.0.0**
* Operating System+version: **Linux Ubuntu 20.04**
* Compiler+version: **GCC 9**
* Conan version: **conan 1.57.0**
* Python version: **Python 3.10.9**


### Conan profile

[settings]
os=Linux
os_build=Linux
arch=x86_64
arch_build=x86_64
build_type=Debug
compiler=gcc
compiler.libcxx=libstdc++11
compiler.version=9
[options]
[conf]
[build_requires]
[env]

### Steps to reproduce

build https://github.com/dvirtz/bugs/tree/conan-arrow-re2
which basically includes this `conanfile.txt`
```
[requires]
arrow/10.0.0

[generators]
CMakeDeps

[options]
arrow:parquet=True
```

and some parquet example code

### Logs

<details><summary>Click to expand log</summary>

```
[main] Building folder: bugs 
[build] Starting build
[proc] Executing command: /home/dyitzchaki/.local/share/virtualenvs/bugs-QWNpjZVn/bin/cmake --build "/home/dyitzchaki/github/bugs/build/GCC 9.4.0 x86_64-linux-gnu_Debug" --config Debug --target all --
[build] [1/1 100% :: 243.455] Linking CXX executable conan-arrow-re2
[build] FAILED: conan-arrow-re2 
[build] : && /usr/bin/g++ -g  CMakeFiles/conan-arrow-re2.dir/reader_writer.cpp.o -o conan-arrow-re2  /home/dyitzchaki/.conan/data/re2/20220601/_/_/package/038baac88f4c7bfa972ce5adac1616bed8fe2ef4/lib/libre2.a  /home/dyitzchaki/.conan/data/arrow/10.0.0/_/_/package/0bc9beab65a90f2058bd00d1dc0cdbc1e897137a/lib/libparquet.a  /home/dyitzchaki/.conan/data/arrow/10.0.0/_/_/package/0bc9beab65a90f2058bd00d1dc0cdbc1e897137a/lib/libarrow.a  /home/dyitzchaki/.conan/data/utf8proc/2.8.0/_/_/package/23b828d52c0630e6b0b96d2945419feb7843c4f8/lib/libutf8proc.a  /home/dyitzchaki/.conan/data/thrift/0.17.0/_/_/package/51186b96cd41af7f403b217cf6f143c4c262b2e3/lib/libthriftnbd.a  /home/dyitzchaki/.conan/data/libevent/2.1.12/_/_/package/b143fdc239af2d3c3b47d49afed8aecee781730b/lib/libevent_pthreads.a  /home/dyitzchaki/.conan/data/libevent/2.1.12/_/_/package/b143fdc239af2d3c3b47d49afed8aecee781730b/lib/libevent_openssl.a  /home/dyitzchaki/.conan/data/libevent/2.1.12/_/_/package/b143fdc239af2d3c3b47d49afed8aecee781730b/lib/libevent_extra.a  /home/dyitzchaki/.conan/data/libevent/2.1.12/_/_/package/b143fdc239af2d3c3b47d49afed8aecee781730b/lib/libevent_core.a  /home/dyitzchaki/.conan/data/thrift/0.17.0/_/_/package/51186b96cd41af7f403b217cf6f143c4c262b2e3/lib/libthriftzd.a  /home/dyitzchaki/.conan/data/zlib/1.2.13/_/_/package/23b828d52c0630e6b0b96d2945419feb7843c4f8/lib/libz.a  /home/dyitzchaki/.conan/data/thrift/0.17.0/_/_/package/51186b96cd41af7f403b217cf6f143c4c262b2e3/lib/libthriftd.a  -lm  /home/dyitzchaki/.conan/data/openssl/1.1.1q/_/_/package/23b828d52c0630e6b0b96d2945419feb7843c4f8/lib/libssl.a  /home/dyitzchaki/.conan/data/openssl/1.1.1q/_/_/package/23b828d52c0630e6b0b96d2945419feb7843c4f8/lib/libcrypto.a  -lpthread  -ldl  -lrt && :
[build] /usr/bin/ld: /home/dyitzchaki/.conan/data/arrow/10.0.0/_/_/package/0bc9beab65a90f2058bd00d1dc0cdbc1e897137a/lib/libarrow.a(scalar_string_ascii.cc.o): in function `arrow::compute::internal::(anonymous namespace)::MakeRE2Options(bool, bool, bool)':
[build] /home/dyitzchaki/.conan/data/arrow/10.0.0/_/_/build/0bc9beab65a90f2058bd00d1dc0cdbc1e897137a/src/cpp/src/arrow/compute/kernels/scalar_string_ascii.cc:63: undefined reference to `re2::RE2::Options::Options(re2::RE2::CannedOptions)'
[build] /usr/bin/ld: /home/dyitzchaki/.conan/data/arrow/10.0.0/_/_/package/0bc9beab65a90f2058bd00d1dc0cdbc1e897137a/lib/libarrow.a(scalar_string_ascii.cc.o): in function `arrow::compute::internal::(anonymous namespace)::RegexSubstringMatcher::RegexSubstringMatcher(arrow::compute::MatchSubstringOptions const&, bool, bool)':
[build] /home/dyitzchaki/.conan/data/arrow/10.0.0/_/_/build/0bc9beab65a90f2058bd00d1dc0cdbc1e897137a/src/cpp/src/arrow/compute/kernels/scalar_string_ascii.cc:1320: undefined reference to `re2::RE2::RE2(re2::StringPiece const&, re2::RE2::Options const&)'
[build] /usr/bin/ld: /home/dyitzchaki/.conan/data/arrow/10.0.0/_/_/package/0bc9beab65a90f2058bd00d1dc0cdbc1e897137a/lib/libarrow.a(scalar_string_ascii.cc.o): in function `arrow::compute::internal::(anonymous namespace)::FindSubstringRegex::FindSubstringRegex(arrow::compute::MatchSubstringOptions const&, bool, bool)':
[build] /home/dyitzchaki/.conan/data/arrow/10.0.0/_/_/build/0bc9beab65a90f2058bd00d1dc0cdbc1e897137a/src/cpp/src/arrow/compute/kernels/scalar_string_ascii.cc:1676: undefined reference to `re2::RE2::QuoteMeta[abi:cxx11](re2::StringPiece const&)'
[build] /usr/bin/ld: /home/dyitzchaki/.conan/data/arrow/10.0.0/_/_/build/0bc9beab65a90f2058bd00d1dc0cdbc1e897137a/src/cpp/src/arrow/compute/kernels/scalar_string_ascii.cc:1679: undefined reference to `re2::RE2::RE2(re2::StringPiece const&, re2::RE2::Options const&)'
[build] /usr/bin/ld: /home/dyitzchaki/.conan/data/arrow/10.0.0/_/_/package/0bc9beab65a90f2058bd00d1dc0cdbc1e897137a/lib/libarrow.a(scalar_string_ascii.cc.o): in function `arrow::compute::internal::(anonymous namespace)::CountSubstringRegex::CountSubstringRegex(arrow::compute::MatchSubstringOptions const&, bool, bool)':
[build] /home/dyitzchaki/.conan/data/arrow/10.0.0/_/_/build/0bc9beab65a90f2058bd00d1dc0cdbc1e897137a/src/cpp/src/arrow/compute/kernels/scalar_string_ascii.cc:1808: undefined reference to `re2::RE2::RE2(re2::StringPiece const&, re2::RE2::Options const&)'
[build] /usr/bin/ld: /home/dyitzchaki/.conan/data/arrow/10.0.0/_/_/package/0bc9beab65a90f2058bd00d1dc0cdbc1e897137a/lib/libarrow.a(scalar_string_ascii.cc.o): in function `arrow::compute::internal::(anonymous namespace)::ExtractRegexData::Make(arrow::compute::ExtractRegexOptions const&, bool)':
[build] /home/dyitzchaki/.conan/data/arrow/10.0.0/_/_/build/0bc9beab65a90f2058bd00d1dc0cdbc1e897137a/src/cpp/src/arrow/compute/kernels/scalar_string_ascii.cc:2171: undefined reference to `re2::RE2::CapturingGroupNames[abi:cxx11]() const'
[build] /usr/bin/ld: /home/dyitzchaki/.conan/data/arrow/10.0.0/_/_/package/0bc9beab65a90f2058bd00d1dc0cdbc1e897137a/lib/libarrow.a(scalar_string_ascii.cc.o): in function `arrow::compute::internal::(anonymous namespace)::ExtractRegexData::ExtractRegexData(std::__cxx11::basic_string<char, std::char_traits<char>, std::allocator<char> > const&, bool)':
[build] /home/dyitzchaki/.conan/data/arrow/10.0.0/_/_/build/0bc9beab65a90f2058bd00d1dc0cdbc1e897137a/src/cpp/src/arrow/compute/kernels/scalar_string_ascii.cc:2204: undefined reference to `re2::RE2::RE2(re2::StringPiece const&, re2::RE2::Options const&)'
[build] /usr/bin/ld: /home/dyitzchaki/.conan/data/arrow/10.0.0/_/_/package/0bc9beab65a90f2058bd00d1dc0cdbc1e897137a/lib/libarrow.a(scalar_string_ascii.cc.o): in function `arrow::compute::internal::(anonymous namespace)::ExtractRegexBase::Match(std::basic_string_view<char, std::char_traits<char> >)':
[build] /home/dyitzchaki/.conan/data/arrow/10.0.0/_/_/build/0bc9beab65a90f2058bd00d1dc0cdbc1e897137a/src/cpp/src/arrow/compute/kernels/scalar_string_ascii.cc:2240: undefined reference to `re2::RE2::PartialMatchN(re2::StringPiece const&, re2::RE2 const&, re2::RE2::Arg const* const*, int)'
[build] /usr/bin/ld: /home/dyitzchaki/.conan/data/arrow/10.0.0/_/_/package/0bc9beab65a90f2058bd00d1dc0cdbc1e897137a/lib/libarrow.a(scalar_string_ascii.cc.o): in function `arrow::compute::internal::(anonymous namespace)::RegexSubstringMatcher::~RegexSubstringMatcher()':
[build] /home/dyitzchaki/.conan/data/arrow/10.0.0/_/_/build/0bc9beab65a90f2058bd00d1dc0cdbc1e897137a/src/cpp/src/arrow/compute/kernels/scalar_string_ascii.cc:1305: undefined reference to `re2::RE2::~RE2()'
[build] /usr/bin/ld: /home/dyitzchaki/.conan/data/arrow/10.0.0/_/_/package/0bc9beab65a90f2058bd00d1dc0cdbc1e897137a/lib/libarrow.a(scalar_string_ascii.cc.o): in function `arrow::compute::internal::(anonymous namespace)::MatchSubstring<arrow::BinaryType, arrow::compute::internal::(anonymous namespace)::PlainStartsWithMatcher>::Exec(arrow::compute::KernelContext*, arrow::compute::ExecSpan const&, arrow::compute::ExecResult*)':
[build] /home/dyitzchaki/.conan/data/arrow/10.0.0/_/_/build/0bc9beab65a90f2058bd00d1dc0cdbc1e897137a/src/cpp/src/arrow/compute/kernels/scalar_string_ascii.cc:1407: undefined reference to `re2::RE2::QuoteMeta[abi:cxx11](re2::StringPiece const&)'
[build] /usr/bin/ld: /home/dyitzchaki/.conan/data/arrow/10.0.0/_/_/package/0bc9beab65a90f2058bd00d1dc0cdbc1e897137a/lib/libarrow.a(scalar_string_ascii.cc.o): in function `arrow::compute::internal::(anonymous namespace)::MatchSubstring<arrow::StringType, arrow::compute::internal::(anonymous namespace)::PlainStartsWithMatcher>::Exec(arrow::compute::KernelContext*, arrow::compute::ExecSpan const&, arrow::compute::ExecResult*)':
[build] /home/dyitzchaki/.conan/data/arrow/10.0.0/_/_/build/0bc9beab65a90f2058bd00d1dc0cdbc1e897137a/src/cpp/src/arrow/compute/kernels/scalar_string_ascii.cc:1407: undefined reference to `re2::RE2::QuoteMeta[abi:cxx11](re2::StringPiece const&)'
[build] /usr/bin/ld: /home/dyitzchaki/.conan/data/arrow/10.0.0/_/_/package/0bc9beab65a90f2058bd00d1dc0cdbc1e897137a/lib/libarrow.a(scalar_string_ascii.cc.o): in function `arrow::compute::internal::(anonymous namespace)::MatchSubstring<arrow::LargeBinaryType, arrow::compute::internal::(anonymous namespace)::PlainStartsWithMatcher>::Exec(arrow::compute::KernelContext*, arrow::compute::ExecSpan const&, arrow::compute::ExecResult*)':
[build] /home/dyitzchaki/.conan/data/arrow/10.0.0/_/_/build/0bc9beab65a90f2058bd00d1dc0cdbc1e897137a/src/cpp/src/arrow/compute/kernels/scalar_string_ascii.cc:1407: undefined reference to `re2::RE2::QuoteMeta[abi:cxx11](re2::StringPiece const&)'
[build] /usr/bin/ld: /home/dyitzchaki/.conan/data/arrow/10.0.0/_/_/package/0bc9beab65a90f2058bd00d1dc0cdbc1e897137a/lib/libarrow.a(scalar_string_ascii.cc.o): in function `arrow::compute::internal::(anonymous namespace)::MatchSubstring<arrow::LargeStringType, arrow::compute::internal::(anonymous namespace)::PlainStartsWithMatcher>::Exec(arrow::compute::KernelContext*, arrow::compute::ExecSpan const&, arrow::compute::ExecResult*)':
[build] /home/dyitzchaki/.conan/data/arrow/10.0.0/_/_/build/0bc9beab65a90f2058bd00d1dc0cdbc1e897137a/src/cpp/src/arrow/compute/kernels/scalar_string_ascii.cc:1407: undefined reference to `re2::RE2::QuoteMeta[abi:cxx11](re2::StringPiece const&)'
[build] /usr/bin/ld: /home/dyitzchaki/.conan/data/arrow/10.0.0/_/_/package/0bc9beab65a90f2058bd00d1dc0cdbc1e897137a/lib/libarrow.a(scalar_string_ascii.cc.o): in function `arrow::compute::internal::(anonymous namespace)::MatchSubstring<arrow::BinaryType, arrow::compute::internal::(anonymous namespace)::PlainEndsWithMatcher>::Exec(arrow::compute::KernelContext*, arrow::compute::ExecSpan const&, arrow::compute::ExecResult*)':
[build] /home/dyitzchaki/.conan/data/arrow/10.0.0/_/_/build/0bc9beab65a90f2058bd00d1dc0cdbc1e897137a/src/cpp/src/arrow/compute/kernels/scalar_string_ascii.cc:1430: undefined reference to `re2::RE2::QuoteMeta[abi:cxx11](re2::StringPiece const&)'
[build] /usr/bin/ld: /home/dyitzchaki/.conan/data/arrow/10.0.0/_/_/package/0bc9beab65a90f2058bd00d1dc0cdbc1e897137a/lib/libarrow.a(scalar_string_ascii.cc.o):/home/dyitzchaki/.conan/data/arrow/10.0.0/_/_/build/0bc9beab65a90f2058bd00d1dc0cdbc1e897137a/src/cpp/src/arrow/compute/kernels/scalar_string_ascii.cc:1430: more undefined references to `re2::RE2::QuoteMeta[abi:cxx11](re2::StringPiece const&)' follow
[build] /usr/bin/ld: /home/dyitzchaki/.conan/data/arrow/10.0.0/_/_/package/0bc9beab65a90f2058bd00d1dc0cdbc1e897137a/lib/libarrow.a(scalar_string_ascii.cc.o): in function `arrow::compute::internal::(anonymous namespace)::RegexSubstringReplacer<arrow::BinaryType>::ReplaceString(std::basic_string_view<char, std::char_traits<char> >, arrow::TypedBufferBuilder<unsigned char, void>*) const':
[build] /home/dyitzchaki/.conan/data/arrow/10.0.0/_/_/build/0bc9beab65a90f2058bd00d1dc0cdbc1e897137a/src/cpp/src/arrow/compute/kernels/scalar_string_ascii.cc:2056: undefined reference to `re2::RE2::GlobalReplace(std::__cxx11::basic_string<char, std::char_traits<char>, std::allocator<char> >*, re2::RE2 const&, re2::StringPiece const&)'
[build] /usr/bin/ld: /home/dyitzchaki/.conan/data/arrow/10.0.0/_/_/build/0bc9beab65a90f2058bd00d1dc0cdbc1e897137a/src/cpp/src/arrow/compute/kernels/scalar_string_ascii.cc:2082: undefined reference to `re2::RE2::Replace(std::__cxx11::basic_string<char, std::char_traits<char>, std::allocator<char> >*, re2::RE2 const&, re2::StringPiece const&)'
[build] /usr/bin/ld: /home/dyitzchaki/.conan/data/arrow/10.0.0/_/_/package/0bc9beab65a90f2058bd00d1dc0cdbc1e897137a/lib/libarrow.a(scalar_string_ascii.cc.o): in function `arrow::compute::internal::(anonymous namespace)::RegexSubstringReplacer<arrow::StringType>::ReplaceString(std::basic_string_view<char, std::char_traits<char> >, arrow::TypedBufferBuilder<unsigned char, void>*) const':
[build] /home/dyitzchaki/.conan/data/arrow/10.0.0/_/_/build/0bc9beab65a90f2058bd00d1dc0cdbc1e897137a/src/cpp/src/arrow/compute/kernels/scalar_string_ascii.cc:2056: undefined reference to `re2::RE2::GlobalReplace(std::__cxx11::basic_string<char, std::char_traits<char>, std::allocator<char> >*, re2::RE2 const&, re2::StringPiece const&)'
[build] /usr/bin/ld: /home/dyitzchaki/.conan/data/arrow/10.0.0/_/_/build/0bc9beab65a90f2058bd00d1dc0cdbc1e897137a/src/cpp/src/arrow/compute/kernels/scalar_string_ascii.cc:2082: undefined reference to `re2::RE2::Replace(std::__cxx11::basic_string<char, std::char_traits<char>, std::allocator<char> >*, re2::RE2 const&, re2::StringPiece const&)'
[build] /usr/bin/ld: /home/dyitzchaki/.conan/data/arrow/10.0.0/_/_/package/0bc9beab65a90f2058bd00d1dc0cdbc1e897137a/lib/libarrow.a(scalar_string_ascii.cc.o): in function `arrow::compute::internal::(anonymous namespace)::RegexSubstringReplacer<arrow::LargeBinaryType>::ReplaceString(std::basic_string_view<char, std::char_traits<char> >, arrow::TypedBufferBuilder<unsigned char, void>*) const':
[build] /home/dyitzchaki/.conan/data/arrow/10.0.0/_/_/build/0bc9beab65a90f2058bd00d1dc0cdbc1e897137a/src/cpp/src/arrow/compute/kernels/scalar_string_ascii.cc:2056: undefined reference to `re2::RE2::GlobalReplace(std::__cxx11::basic_string<char, std::char_traits<char>, std::allocator<char> >*, re2::RE2 const&, re2::StringPiece const&)'
[build] /usr/bin/ld: /home/dyitzchaki/.conan/data/arrow/10.0.0/_/_/build/0bc9beab65a90f2058bd00d1dc0cdbc1e897137a/src/cpp/src/arrow/compute/kernels/scalar_string_ascii.cc:2082: undefined reference to `re2::RE2::Replace(std::__cxx11::basic_string<char, std::char_traits<char>, std::allocator<char> >*, re2::RE2 const&, re2::StringPiece const&)'
[build] /usr/bin/ld: /home/dyitzchaki/.conan/data/arrow/10.0.0/_/_/package/0bc9beab65a90f2058bd00d1dc0cdbc1e897137a/lib/libarrow.a(scalar_string_ascii.cc.o): in function `arrow::compute::internal::(anonymous namespace)::RegexSubstringReplacer<arrow::LargeStringType>::ReplaceString(std::basic_string_view<char, std::char_traits<char> >, arrow::TypedBufferBuilder<unsigned char, void>*) const':
[build] /home/dyitzchaki/.conan/data/arrow/10.0.0/_/_/build/0bc9beab65a90f2058bd00d1dc0cdbc1e897137a/src/cpp/src/arrow/compute/kernels/scalar_string_ascii.cc:2056: undefined reference to `re2::RE2::GlobalReplace(std::__cxx11::basic_string<char, std::char_traits<char>, std::allocator<char> >*, re2::RE2 const&, re2::StringPiece const&)'
[build] /usr/bin/ld: /home/dyitzchaki/.conan/data/arrow/10.0.0/_/_/build/0bc9beab65a90f2058bd00d1dc0cdbc1e897137a/src/cpp/src/arrow/compute/kernels/scalar_string_ascii.cc:2082: undefined reference to `re2::RE2::Replace(std::__cxx11::basic_string<char, std::char_traits<char>, std::allocator<char> >*, re2::RE2 const&, re2::StringPiece const&)'
[build] /usr/bin/ld: /home/dyitzchaki/.conan/data/arrow/10.0.0/_/_/package/0bc9beab65a90f2058bd00d1dc0cdbc1e897137a/lib/libarrow.a(scalar_string_ascii.cc.o): in function `bool re2::RE2::PartialMatch<>(re2::StringPiece const&, re2::RE2 const&)':
[build] /home/dyitzchaki/.conan/data/re2/20220601/_/_/package/038baac88f4c7bfa972ce5adac1616bed8fe2ef4/include/re2/re2.h:404: undefined reference to `re2::RE2::PartialMatchN(re2::StringPiece const&, re2::RE2 const&, re2::RE2::Arg const* const*, int)'
[build] /usr/bin/ld: /home/dyitzchaki/.conan/data/arrow/10.0.0/_/_/package/0bc9beab65a90f2058bd00d1dc0cdbc1e897137a/lib/libarrow.a(scalar_string_ascii.cc.o): in function `arrow::compute::internal::(anonymous namespace)::MatchLike<arrow::BinaryType>::Exec(arrow::compute::KernelContext*, arrow::compute::ExecSpan const&, arrow::compute::ExecResult*)':
[build] /home/dyitzchaki/.conan/data/arrow/10.0.0/_/_/build/0bc9beab65a90f2058bd00d1dc0cdbc1e897137a/src/cpp/src/arrow/compute/kernels/scalar_string_ascii.cc:1503: undefined reference to `re2::RE2::RE2(re2::StringPiece const&, re2::RE2::Options const&)'
[build] /usr/bin/ld: /home/dyitzchaki/.conan/data/arrow/10.0.0/_/_/build/0bc9beab65a90f2058bd00d1dc0cdbc1e897137a/src/cpp/src/arrow/compute/kernels/scalar_string_ascii.cc:1503: undefined reference to `re2::RE2::~RE2()'
[build] /usr/bin/ld: /home/dyitzchaki/.conan/data/arrow/10.0.0/_/_/build/0bc9beab65a90f2058bd00d1dc0cdbc1e897137a/src/cpp/src/arrow/compute/kernels/scalar_string_ascii.cc:1505: undefined reference to `re2::RE2::RE2(re2::StringPiece const&, re2::RE2::Options const&)'
[build] /usr/bin/ld: /home/dyitzchaki/.conan/data/arrow/10.0.0/_/_/build/0bc9beab65a90f2058bd00d1dc0cdbc1e897137a/src/cpp/src/arrow/compute/kernels/scalar_string_ascii.cc:1505: undefined reference to `re2::RE2::~RE2()'
[build] /usr/bin/ld: /home/dyitzchaki/.conan/data/arrow/10.0.0/_/_/build/0bc9beab65a90f2058bd00d1dc0cdbc1e897137a/src/cpp/src/arrow/compute/kernels/scalar_string_ascii.cc:1507: undefined reference to `re2::RE2::RE2(re2::StringPiece const&, re2::RE2::Options const&)'
[build] /usr/bin/ld: /home/dyitzchaki/.conan/data/arrow/10.0.0/_/_/build/0bc9beab65a90f2058bd00d1dc0cdbc1e897137a/src/cpp/src/arrow/compute/kernels/scalar_string_ascii.cc:1507: undefined reference to `re2::RE2::~RE2()'
[build] /usr/bin/ld: /home/dyitzchaki/.conan/data/arrow/10.0.0/_/_/package/0bc9beab65a90f2058bd00d1dc0cdbc1e897137a/lib/libarrow.a(scalar_string_ascii.cc.o): in function `arrow::compute::internal::(anonymous namespace)::MatchLike<arrow::StringType>::Exec(arrow::compute::KernelContext*, arrow::compute::ExecSpan const&, arrow::compute::ExecResult*)':
[build] /home/dyitzchaki/.conan/data/arrow/10.0.0/_/_/build/0bc9beab65a90f2058bd00d1dc0cdbc1e897137a/src/cpp/src/arrow/compute/kernels/scalar_string_ascii.cc:1503: undefined reference to `re2::RE2::RE2(re2::StringPiece const&, re2::RE2::Options const&)'
[build] /usr/bin/ld: /home/dyitzchaki/.conan/data/arrow/10.0.0/_/_/build/0bc9beab65a90f2058bd00d1dc0cdbc1e897137a/src/cpp/src/arrow/compute/kernels/scalar_string_ascii.cc:1503: undefined reference to `re2::RE2::~RE2()'
[build] /usr/bin/ld: /home/dyitzchaki/.conan/data/arrow/10.0.0/_/_/build/0bc9beab65a90f2058bd00d1dc0cdbc1e897137a/src/cpp/src/arrow/compute/kernels/scalar_string_ascii.cc:1505: undefined reference to `re2::RE2::RE2(re2::StringPiece const&, re2::RE2::Options const&)'
[build] /usr/bin/ld: /home/dyitzchaki/.conan/data/arrow/10.0.0/_/_/build/0bc9beab65a90f2058bd00d1dc0cdbc1e897137a/src/cpp/src/arrow/compute/kernels/scalar_string_ascii.cc:1505: undefined reference to `re2::RE2::~RE2()'
[build] /usr/bin/ld: /home/dyitzchaki/.conan/data/arrow/10.0.0/_/_/build/0bc9beab65a90f2058bd00d1dc0cdbc1e897137a/src/cpp/src/arrow/compute/kernels/scalar_string_ascii.cc:1507: undefined reference to `re2::RE2::RE2(re2::StringPiece const&, re2::RE2::Options const&)'
[build] /usr/bin/ld: /home/dyitzchaki/.conan/data/arrow/10.0.0/_/_/build/0bc9beab65a90f2058bd00d1dc0cdbc1e897137a/src/cpp/src/arrow/compute/kernels/scalar_string_ascii.cc:1507: undefined reference to `re2::RE2::~RE2()'
[build] /usr/bin/ld: /home/dyitzchaki/.conan/data/arrow/10.0.0/_/_/package/0bc9beab65a90f2058bd00d1dc0cdbc1e897137a/lib/libarrow.a(scalar_string_ascii.cc.o): in function `arrow::compute::internal::(anonymous namespace)::MatchLike<arrow::LargeBinaryType>::Exec(arrow::compute::KernelContext*, arrow::compute::ExecSpan const&, arrow::compute::ExecResult*)':
[build] /home/dyitzchaki/.conan/data/arrow/10.0.0/_/_/build/0bc9beab65a90f2058bd00d1dc0cdbc1e897137a/src/cpp/src/arrow/compute/kernels/scalar_string_ascii.cc:1503: undefined reference to `re2::RE2::RE2(re2::StringPiece const&, re2::RE2::Options const&)'
[build] /usr/bin/ld: /home/dyitzchaki/.conan/data/arrow/10.0.0/_/_/build/0bc9beab65a90f2058bd00d1dc0cdbc1e897137a/src/cpp/src/arrow/compute/kernels/scalar_string_ascii.cc:1503: undefined reference to `re2::RE2::~RE2()'
[build] /usr/bin/ld: /home/dyitzchaki/.conan/data/arrow/10.0.0/_/_/build/0bc9beab65a90f2058bd00d1dc0cdbc1e897137a/src/cpp/src/arrow/compute/kernels/scalar_string_ascii.cc:1505: undefined reference to `re2::RE2::RE2(re2::StringPiece const&, re2::RE2::Options const&)'
[build] /usr/bin/ld: /home/dyitzchaki/.conan/data/arrow/10.0.0/_/_/build/0bc9beab65a90f2058bd00d1dc0cdbc1e897137a/src/cpp/src/arrow/compute/kernels/scalar_string_ascii.cc:1505: undefined reference to `re2::RE2::~RE2()'
[build] /usr/bin/ld: /home/dyitzchaki/.conan/data/arrow/10.0.0/_/_/build/0bc9beab65a90f2058bd00d1dc0cdbc1e897137a/src/cpp/src/arrow/compute/kernels/scalar_string_ascii.cc:1507: undefined reference to `re2::RE2::RE2(re2::StringPiece const&, re2::RE2::Options const&)'
[build] /usr/bin/ld: /home/dyitzchaki/.conan/data/arrow/10.0.0/_/_/build/0bc9beab65a90f2058bd00d1dc0cdbc1e897137a/src/cpp/src/arrow/compute/kernels/scalar_string_ascii.cc:1507: undefined reference to `re2::RE2::~RE2()'
[build] /usr/bin/ld: /home/dyitzchaki/.conan/data/arrow/10.0.0/_/_/package/0bc9beab65a90f2058bd00d1dc0cdbc1e897137a/lib/libarrow.a(scalar_string_ascii.cc.o): in function `arrow::compute::internal::(anonymous namespace)::MatchLike<arrow::LargeStringType>::Exec(arrow::compute::KernelContext*, arrow::compute::ExecSpan const&, arrow::compute::ExecResult*)':
[build] /home/dyitzchaki/.conan/data/arrow/10.0.0/_/_/build/0bc9beab65a90f2058bd00d1dc0cdbc1e897137a/src/cpp/src/arrow/compute/kernels/scalar_string_ascii.cc:1503: undefined reference to `re2::RE2::RE2(re2::StringPiece const&, re2::RE2::Options const&)'
[build] /usr/bin/ld: /home/dyitzchaki/.conan/data/arrow/10.0.0/_/_/build/0bc9beab65a90f2058bd00d1dc0cdbc1e897137a/src/cpp/src/arrow/compute/kernels/scalar_string_ascii.cc:1503: undefined reference to `re2::RE2::~RE2()'
[build] /usr/bin/ld: /home/dyitzchaki/.conan/data/arrow/10.0.0/_/_/build/0bc9beab65a90f2058bd00d1dc0cdbc1e897137a/src/cpp/src/arrow/compute/kernels/scalar_string_ascii.cc:1505: undefined reference to `re2::RE2::RE2(re2::StringPiece const&, re2::RE2::Options const&)'
[build] /usr/bin/ld: /home/dyitzchaki/.conan/data/arrow/10.0.0/_/_/build/0bc9beab65a90f2058bd00d1dc0cdbc1e897137a/src/cpp/src/arrow/compute/kernels/scalar_string_ascii.cc:1505: undefined reference to `re2::RE2::~RE2()'
[build] /usr/bin/ld: /home/dyitzchaki/.conan/data/arrow/10.0.0/_/_/build/0bc9beab65a90f2058bd00d1dc0cdbc1e897137a/src/cpp/src/arrow/compute/kernels/scalar_string_ascii.cc:1507: undefined reference to `re2::RE2::RE2(re2::StringPiece const&, re2::RE2::Options const&)'
[build] /usr/bin/ld: /home/dyitzchaki/.conan/data/arrow/10.0.0/_/_/build/0bc9beab65a90f2058bd00d1dc0cdbc1e897137a/src/cpp/src/arrow/compute/kernels/scalar_string_ascii.cc:1507: undefined reference to `re2::RE2::~RE2()'
[build] /usr/bin/ld: /home/dyitzchaki/.conan/data/arrow/10.0.0/_/_/package/0bc9beab65a90f2058bd00d1dc0cdbc1e897137a/lib/libarrow.a(scalar_string_ascii.cc.o): in function `std::default_delete<re2::RE2>::operator()(re2::RE2*) const':
[build] /usr/include/c++/9/bits/unique_ptr.h:81: undefined reference to `re2::RE2::~RE2()'
[build] /usr/bin/ld: /home/dyitzchaki/.conan/data/arrow/10.0.0/_/_/package/0bc9beab65a90f2058bd00d1dc0cdbc1e897137a/lib/libarrow.a(scalar_string_ascii.cc.o): in function `bool re2::RE2::FullMatch<std::__cxx11::basic_string<char, std::char_traits<char>, std::allocator<char> >*>(re2::StringPiece const&, re2::RE2 const&, std::__cxx11::basic_string<char, std::char_traits<char>, std::allocator<char> >*&&)':
[build] /home/dyitzchaki/.conan/data/re2/20220601/_/_/package/038baac88f4c7bfa972ce5adac1616bed8fe2ef4/include/re2/re2.h:388: undefined reference to `re2::RE2::FullMatchN(re2::StringPiece const&, re2::RE2 const&, re2::RE2::Arg const* const*, int)'
[build] /usr/bin/ld: /home/dyitzchaki/.conan/data/arrow/10.0.0/_/_/package/0bc9beab65a90f2058bd00d1dc0cdbc1e897137a/lib/libarrow.a(scalar_string_ascii.cc.o): in function `arrow::compute::internal::(anonymous namespace)::RegexSubstringReplacer<arrow::BinaryType>::Make(arrow::compute::ReplaceSubstringOptions const&)':
[build] /home/dyitzchaki/.conan/data/arrow/10.0.0/_/_/build/0bc9beab65a90f2058bd00d1dc0cdbc1e897137a/src/cpp/src/arrow/compute/kernels/scalar_string_ascii.cc:2031: undefined reference to `re2::RE2::CheckRewriteString(re2::StringPiece const&, std::__cxx11::basic_string<char, std::char_traits<char>, std::allocator<char> >*) const'
[build] /usr/bin/ld: /home/dyitzchaki/.conan/data/arrow/10.0.0/_/_/package/0bc9beab65a90f2058bd00d1dc0cdbc1e897137a/lib/libarrow.a(scalar_string_ascii.cc.o): in function `arrow::compute::internal::(anonymous namespace)::RegexSubstringReplacer<arrow::StringType>::Make(arrow::compute::ReplaceSubstringOptions const&)':
[build] /home/dyitzchaki/.conan/data/arrow/10.0.0/_/_/build/0bc9beab65a90f2058bd00d1dc0cdbc1e897137a/src/cpp/src/arrow/compute/kernels/scalar_string_ascii.cc:2031: undefined reference to `re2::RE2::CheckRewriteString(re2::StringPiece const&, std::__cxx11::basic_string<char, std::char_traits<char>, std::allocator<char> >*) const'
[build] /usr/bin/ld: /home/dyitzchaki/.conan/data/arrow/10.0.0/_/_/package/0bc9beab65a90f2058bd00d1dc0cdbc1e897137a/lib/libarrow.a(scalar_string_ascii.cc.o): in function `arrow::compute::internal::(anonymous namespace)::RegexSubstringReplacer<arrow::LargeBinaryType>::Make(arrow::compute::ReplaceSubstringOptions const&)':
[build] /home/dyitzchaki/.conan/data/arrow/10.0.0/_/_/build/0bc9beab65a90f2058bd00d1dc0cdbc1e897137a/src/cpp/src/arrow/compute/kernels/scalar_string_ascii.cc:2031: undefined reference to `re2::RE2::CheckRewriteString(re2::StringPiece const&, std::__cxx11::basic_string<char, std::char_traits<char>, std::allocator<char> >*) const'
[build] /usr/bin/ld: /home/dyitzchaki/.conan/data/arrow/10.0.0/_/_/package/0bc9beab65a90f2058bd00d1dc0cdbc1e897137a/lib/libarrow.a(scalar_string_ascii.cc.o): in function `arrow::compute::internal::(anonymous namespace)::RegexSubstringReplacer<arrow::LargeStringType>::Make(arrow::compute::ReplaceSubstringOptions const&)':
[build] /home/dyitzchaki/.conan/data/arrow/10.0.0/_/_/build/0bc9beab65a90f2058bd00d1dc0cdbc1e897137a/src/cpp/src/arrow/compute/kernels/scalar_string_ascii.cc:2031: undefined reference to `re2::RE2::CheckRewriteString(re2::StringPiece const&, std::__cxx11::basic_string<char, std::char_traits<char>, std::allocator<char> >*) const'
[build] /usr/bin/ld: /home/dyitzchaki/.conan/data/arrow/10.0.0/_/_/package/0bc9beab65a90f2058bd00d1dc0cdbc1e897137a/lib/libarrow.a(scalar_string_ascii.cc.o): in function `arrow::compute::internal::(anonymous namespace)::RegexSubstringReplacer<arrow::BinaryType>::~RegexSubstringReplacer()':
[build] /home/dyitzchaki/.conan/data/arrow/10.0.0/_/_/build/0bc9beab65a90f2058bd00d1dc0cdbc1e897137a/src/cpp/src/arrow/compute/kernels/scalar_string_ascii.cc:2018: undefined reference to `re2::RE2::~RE2()'
[build] /usr/bin/ld: /home/dyitzchaki/.conan/data/arrow/10.0.0/_/_/build/0bc9beab65a90f2058bd00d1dc0cdbc1e897137a/src/cpp/src/arrow/compute/kernels/scalar_string_ascii.cc:2018: undefined reference to `re2::RE2::~RE2()'
[build] /usr/bin/ld: /home/dyitzchaki/.conan/data/arrow/10.0.0/_/_/package/0bc9beab65a90f2058bd00d1dc0cdbc1e897137a/lib/libarrow.a(scalar_string_ascii.cc.o): in function `arrow::compute::internal::(anonymous namespace)::RegexSubstringReplacer<arrow::StringType>::~RegexSubstringReplacer()':
[build] /home/dyitzchaki/.conan/data/arrow/10.0.0/_/_/build/0bc9beab65a90f2058bd00d1dc0cdbc1e897137a/src/cpp/src/arrow/compute/kernels/scalar_string_ascii.cc:2018: undefined reference to `re2::RE2::~RE2()'
[build] /usr/bin/ld: /home/dyitzchaki/.conan/data/arrow/10.0.0/_/_/build/0bc9beab65a90f2058bd00d1dc0cdbc1e897137a/src/cpp/src/arrow/compute/kernels/scalar_string_ascii.cc:2018: undefined reference to `re2::RE2::~RE2()'
[build] /usr/bin/ld: /home/dyitzchaki/.conan/data/arrow/10.0.0/_/_/package/0bc9beab65a90f2058bd00d1dc0cdbc1e897137a/lib/libarrow.a(scalar_string_ascii.cc.o): in function `arrow::compute::internal::(anonymous namespace)::RegexSubstringReplacer<arrow::LargeBinaryType>::~RegexSubstringReplacer()':
[build] /home/dyitzchaki/.conan/data/arrow/10.0.0/_/_/build/0bc9beab65a90f2058bd00d1dc0cdbc1e897137a/src/cpp/src/arrow/compute/kernels/scalar_string_ascii.cc:2018: undefined reference to `re2::RE2::~RE2()'
[build] /usr/bin/ld: /home/dyitzchaki/.conan/data/arrow/10.0.0/_/_/package/0bc9beab65a90f2058bd00d1dc0cdbc1e897137a/lib/libarrow.a(scalar_string_ascii.cc.o):/home/dyitzchaki/.conan/data/arrow/10.0.0/_/_/build/0bc9beab65a90f2058bd00d1dc0cdbc1e897137a/src/cpp/src/arrow/compute/kernels/scalar_string_ascii.cc:2018: more undefined references to `re2::RE2::~RE2()' follow
[build] /usr/bin/ld: /home/dyitzchaki/.conan/data/arrow/10.0.0/_/_/package/0bc9beab65a90f2058bd00d1dc0cdbc1e897137a/lib/libarrow.a(scalar_string_ascii.cc.o): in function `bool re2::RE2::Arg::DoParse3ary<std::__cxx11::basic_string<char, std::char_traits<char>, std::allocator<char> > >(char const*, unsigned long, void*)':
[build] /home/dyitzchaki/.conan/data/re2/20220601/_/_/package/038baac88f4c7bfa972ce5adac1616bed8fe2ef4/include/re2/re2.h:870: undefined reference to `bool re2::re2_internal::Parse<std::__cxx11::basic_string<char, std::char_traits<char>, std::allocator<char> > >(char const*, unsigned long, std::__cxx11::basic_string<char, std::char_traits<char>, std::allocator<char> >*)'
[build] /usr/bin/ld: /home/dyitzchaki/.conan/data/arrow/10.0.0/_/_/package/0bc9beab65a90f2058bd00d1dc0cdbc1e897137a/lib/libarrow.a(scalar_string_ascii.cc.o): in function `bool re2::RE2::PartialMatch<re2::StringPiece*>(re2::StringPiece const&, re2::RE2 const&, re2::StringPiece*&&)':
[build] /home/dyitzchaki/.conan/data/re2/20220601/_/_/package/038baac88f4c7bfa972ce5adac1616bed8fe2ef4/include/re2/re2.h:404: undefined reference to `re2::RE2::PartialMatchN(re2::StringPiece const&, re2::RE2 const&, re2::RE2::Arg const* const*, int)'
[build] /usr/bin/ld: /home/dyitzchaki/.conan/data/arrow/10.0.0/_/_/package/0bc9beab65a90f2058bd00d1dc0cdbc1e897137a/lib/libarrow.a(scalar_string_ascii.cc.o): in function `bool re2::RE2::FindAndConsume<>(re2::StringPiece*, re2::RE2 const&)':
[build] /home/dyitzchaki/.conan/data/re2/20220601/_/_/package/038baac88f4c7bfa972ce5adac1616bed8fe2ef4/include/re2/re2.h:440: undefined reference to `re2::RE2::FindAndConsumeN(re2::StringPiece*, re2::RE2 const&, re2::RE2::Arg const* const*, int)'
[build] /usr/bin/ld: /home/dyitzchaki/.conan/data/arrow/10.0.0/_/_/package/0bc9beab65a90f2058bd00d1dc0cdbc1e897137a/lib/libarrow.a(scalar_string_ascii.cc.o): in function `arrow::compute::internal::(anonymous namespace)::RegexSubstringReplacer<arrow::BinaryType>::RegexSubstringReplacer(arrow::compute::ReplaceSubstringOptions const&)':
[build] /home/dyitzchaki/.conan/data/arrow/10.0.0/_/_/build/0bc9beab65a90f2058bd00d1dc0cdbc1e897137a/src/cpp/src/arrow/compute/kernels/scalar_string_ascii.cc:2045: undefined reference to `re2::RE2::RE2(re2::StringPiece const&, re2::RE2::Options const&)'
[build] /usr/bin/ld: /home/dyitzchaki/.conan/data/arrow/10.0.0/_/_/build/0bc9beab65a90f2058bd00d1dc0cdbc1e897137a/src/cpp/src/arrow/compute/kernels/scalar_string_ascii.cc:2045: undefined reference to `re2::RE2::RE2(re2::StringPiece const&, re2::RE2::Options const&)'
[build] /usr/bin/ld: /home/dyitzchaki/.conan/data/arrow/10.0.0/_/_/build/0bc9beab65a90f2058bd00d1dc0cdbc1e897137a/src/cpp/src/arrow/compute/kernels/scalar_string_ascii.cc:2045: undefined reference to `re2::RE2::~RE2()'
[build] /usr/bin/ld: /home/dyitzchaki/.conan/data/arrow/10.0.0/_/_/package/0bc9beab65a90f2058bd00d1dc0cdbc1e897137a/lib/libarrow.a(scalar_string_ascii.cc.o): in function `bool re2::RE2::FindAndConsume<std::__cxx11::basic_string<char, std::char_traits<char>, std::allocator<char> >*>(re2::StringPiece*, re2::RE2 const&, std::__cxx11::basic_string<char, std::char_traits<char>, std::allocator<char> >*&&)':
[build] /home/dyitzchaki/.conan/data/re2/20220601/_/_/package/038baac88f4c7bfa972ce5adac1616bed8fe2ef4/include/re2/re2.h:440: undefined reference to `re2::RE2::FindAndConsumeN(re2::StringPiece*, re2::RE2 const&, re2::RE2::Arg const* const*, int)'
[build] /usr/bin/ld: /home/dyitzchaki/.conan/data/arrow/10.0.0/_/_/package/0bc9beab65a90f2058bd00d1dc0cdbc1e897137a/lib/libarrow.a(scalar_string_ascii.cc.o): in function `arrow::compute::internal::(anonymous namespace)::RegexSubstringReplacer<arrow::StringType>::RegexSubstringReplacer(arrow::compute::ReplaceSubstringOptions const&)':
[build] /home/dyitzchaki/.conan/data/arrow/10.0.0/_/_/build/0bc9beab65a90f2058bd00d1dc0cdbc1e897137a/src/cpp/src/arrow/compute/kernels/scalar_string_ascii.cc:2045: undefined reference to `re2::RE2::RE2(re2::StringPiece const&, re2::RE2::Options const&)'
[build] /usr/bin/ld: /home/dyitzchaki/.conan/data/arrow/10.0.0/_/_/build/0bc9beab65a90f2058bd00d1dc0cdbc1e897137a/src/cpp/src/arrow/compute/kernels/scalar_string_ascii.cc:2045: undefined reference to `re2::RE2::RE2(re2::StringPiece const&, re2::RE2::Options const&)'
[build] /usr/bin/ld: /home/dyitzchaki/.conan/data/arrow/10.0.0/_/_/build/0bc9beab65a90f2058bd00d1dc0cdbc1e897137a/src/cpp/src/arrow/compute/kernels/scalar_string_ascii.cc:2045: undefined reference to `re2::RE2::~RE2()'
[build] /usr/bin/ld: /home/dyitzchaki/.conan/data/arrow/10.0.0/_/_/package/0bc9beab65a90f2058bd00d1dc0cdbc1e897137a/lib/libarrow.a(scalar_string_ascii.cc.o): in function `arrow::compute::internal::(anonymous namespace)::RegexSubstringReplacer<arrow::LargeBinaryType>::RegexSubstringReplacer(arrow::compute::ReplaceSubstringOptions const&)':
[build] /home/dyitzchaki/.conan/data/arrow/10.0.0/_/_/build/0bc9beab65a90f2058bd00d1dc0cdbc1e897137a/src/cpp/src/arrow/compute/kernels/scalar_string_ascii.cc:2045: undefined reference to `re2::RE2::RE2(re2::StringPiece const&, re2::RE2::Options const&)'
[build] /usr/bin/ld: /home/dyitzchaki/.conan/data/arrow/10.0.0/_/_/build/0bc9beab65a90f2058bd00d1dc0cdbc1e897137a/src/cpp/src/arrow/compute/kernels/scalar_string_ascii.cc:2045: undefined reference to `re2::RE2::RE2(re2::StringPiece const&, re2::RE2::Options const&)'
[build] /usr/bin/ld: /home/dyitzchaki/.conan/data/arrow/10.0.0/_/_/build/0bc9beab65a90f2058bd00d1dc0cdbc1e897137a/src/cpp/src/arrow/compute/kernels/scalar_string_ascii.cc:2045: undefined reference to `re2::RE2::~RE2()'
[build] /usr/bin/ld: /home/dyitzchaki/.conan/data/arrow/10.0.0/_/_/package/0bc9beab65a90f2058bd00d1dc0cdbc1e897137a/lib/libarrow.a(scalar_string_ascii.cc.o): in function `arrow::compute::internal::(anonymous namespace)::RegexSubstringReplacer<arrow::LargeStringType>::RegexSubstringReplacer(arrow::compute::ReplaceSubstringOptions const&)':
[build] /home/dyitzchaki/.conan/data/arrow/10.0.0/_/_/build/0bc9beab65a90f2058bd00d1dc0cdbc1e897137a/src/cpp/src/arrow/compute/kernels/scalar_string_ascii.cc:2045: undefined reference to `re2::RE2::RE2(re2::StringPiece const&, re2::RE2::Options const&)'
[build] /usr/bin/ld: /home/dyitzchaki/.conan/data/arrow/10.0.0/_/_/build/0bc9beab65a90f2058bd00d1dc0cdbc1e897137a/src/cpp/src/arrow/compute/kernels/scalar_string_ascii.cc:2045: undefined reference to `re2::RE2::RE2(re2::StringPiece const&, re2::RE2::Options const&)'
[build] /usr/bin/ld: /home/dyitzchaki/.conan/data/arrow/10.0.0/_/_/build/0bc9beab65a90f2058bd00d1dc0cdbc1e897137a/src/cpp/src/arrow/compute/kernels/scalar_string_ascii.cc:2045: undefined reference to `re2::RE2::~RE2()'
[build] /usr/bin/ld: /home/dyitzchaki/.conan/data/arrow/10.0.0/_/_/package/0bc9beab65a90f2058bd00d1dc0cdbc1e897137a/lib/libarrow.a(scalar_string_ascii.cc.o): in function `bool re2::RE2::Arg::DoParse3ary<re2::StringPiece>(char const*, unsigned long, void*)':
[build] /home/dyitzchaki/.conan/data/re2/20220601/_/_/package/038baac88f4c7bfa972ce5adac1616bed8fe2ef4/include/re2/re2.h:870: undefined reference to `bool re2::re2_internal::Parse<re2::StringPiece>(char const*, unsigned long, re2::StringPiece*)'
[build] /usr/bin/ld: /home/dyitzchaki/.conan/data/arrow/10.0.0/_/_/package/0bc9beab65a90f2058bd00d1dc0cdbc1e897137a/lib/libarrow.a(scalar_string_ascii.cc.o): in function `std::_MakeUniq<re2::RE2>::__single_object std::make_unique<re2::RE2, std::__cxx11::basic_string<char, std::char_traits<char>, std::allocator<char> >&, re2::RE2::Options>(std::__cxx11::basic_string<char, std::char_traits<char>, std::allocator<char> >&, re2::RE2::Options&&)':
[build] /usr/include/c++/9/bits/unique_ptr.h:857: undefined reference to `re2::RE2::RE2(re2::StringPiece const&, re2::RE2::Options const&)'
[build] collect2: error: ld returned 1 exit status
[build] ninja: build stopped: subcommand failed.
[proc] The command: /home/dyitzchaki/.local/share/virtualenvs/bugs-QWNpjZVn/bin/cmake --build "/home/dyitzchaki/github/bugs/build/GCC 9.4.0 x86_64-linux-gnu_Debug" --config Debug --target all -- exited with code: 1
[build] Build finished with exit code 1
```

</details>

--- a/recipes/arrow/all/conanfile.py
+++ b/recipes/arrow/all/conanfile.py
@@ -637,7 +637,8 @@ class ArrowConan(ConanFile):
         if self.options.with_mimalloc:
             self.cpp_info.components["libarrow"].requires.append("mimalloc::mimalloc")
         if self._with_re2():
-            self.cpp_info.components["libgandiva"].requires.append("re2::re2")
+            for comp in ["libgandiva", "libparquet", "libarrow"]:
+                self.cpp_info.components[comp].requires.append("re2::re2")
         if self._with_llvm():
             self.cpp_info.components["libgandiva"].requires.append("llvm-core::llvm-core")
         if self._with_protobuf():

--- a/recipes/arrow/all/conanfile.py
+++ b/recipes/arrow/all/conanfile.py
@@ -637,8 +637,11 @@ class ArrowConan(ConanFile):
         if self.options.with_mimalloc:
             self.cpp_info.components["libarrow"].requires.append("mimalloc::mimalloc")
         if self._with_re2():
-            for comp in ["libgandiva", "libparquet", "libarrow"]:
-                self.cpp_info.components[comp].requires.append("re2::re2")
+            if self.options.gandiva:
+                self.cpp_info.components["libgandiva"].requires.append("re2::re2")
+            if self._parquet():
+                self.cpp_info.components["libparquet"].requires.append("re2::re2")
+            self.cpp_info.components["libarrow"].requires.append("re2::re2")
         if self._with_llvm():
             self.cpp_info.components["libgandiva"].requires.append("llvm-core::llvm-core")
         if self._with_protobuf():

